### PR TITLE
Nuclear Self Destruct [Now with added centcomm]

### PR DESCRIPTION
**Pull request by Dorsisdwarf**
This adds the nuclear self destruct to the game (Not the station, since I cannot map-fu). This is an anchored version of the nuclear bomb, which cannot be unanchored via the nuke panel.

The idea behind this is so that you don't have to have an actual stealable nuclear bomb in the station, and implements a proper nuclear self-destruct system other than "Beg admins for code, plug into nuke".
Namely, beg them for the code with a special button, get BSA'd if they don't like you.

Obligatory in-game image:
herocaptain

Active nuke:
bignuk1

Ticking down...  
bignuk2

### CODE REVIEW

**Comment by Jalleo** [comment]
How will the destruct sequence be activated and does this egfect malf AI?

**Comment by Alek2ander** [comment]
rip clown stealing the nuke shenanigans

**Comment by Dorsisdwarf** [comment]
This effects malf in no manner whatsoever. The vault nuke doesn't actually have anything to do with the station self-destruct codewise, which is why making it spaceable and on a trolley of all things is dumb. 

As it is, this is a varedited nuke with a nice nuke detonation animation that cannot be moved under any circumstances. Next up is getting it on the map, after which introducing the nuke call.

**Comment by Deantwo** [comment]
Helps if you put a link to the forum thread in the opening post.
Read this:https://tgstation13.org/phpBB/viewtopic.php?f=9&t=2774

**Comment by TheVekter** [comment]
My idea is getting implemented, doushio

:+1: EXCEPT I don't like it being in the vault. Everyone knows that place isn't REALLY secure. I almost want a different room for it, but oh well.

**Comment by Dorsisdwarf** [comment]
I'm kinda tempted to replace the captain's bathroom with this (But keep it named the bathroom), so the AI has to treat it like the execution room (can't metagame the fact they know it's what makes everything boom)

**Comment by TheVekter** [comment]
I like that. Makes it hard to find.

**Comment by Dorsisdwarf** [comment]
Updated OP with actual animations

**Comment by TheVekter** [comment]
@Jalleo: It's a nuke but always anchored and with new cool sprites. Still have to ask admins for the key, still can't actually set it off without admin intervention.

**Comment by Kromgar** [comment]
Make the nuke work with delta now. If AI fucks up and lets  science gets BoH's they could destroy the nuke and the AI can only take full control of station

**Comment by Dorsisdwarf** [comment]
The station nuke is still a trolley-based-object currently, since I really dont want to figure out how to merge maps or whatever the mapping process is.

**Comment by Dorsisdwarf** [comment]
UPDATE: You can now use either the captain's ID at a comms console to call command and ask for the codes. This sends a centcomm announcement to the crew which shouts your name to them and says what's going on, then pings the admins and asks them for the code (Reason must be included). BSA is included as a HREF in order to punish the frivolous.

**Comment by TheVekter** [comment]
>BSA is included as a HREF in order to punish the frivolous.

Good. This needs to be an option more often.

**Comment by Dorsisdwarf** [comment]
I'm not sure what's going on, but calling from the authenticator is supposed to be disabled as of commit "removes authenticator", and it isn't

**Comment by Deantwo** [comment]
Yeah sorry, I was using my phone and didn't notice I was accidentally commenting on the  https://github.com/Dorsisdwarf/-tg-station/commit/b234b0dc860eeeaaf205608ffdb90f4b6bb71cb6 commit rather than the PR's file changes.
But still.

**Comment by Dorsisdwarf** [comment]
Because those were commented onto the commit, they dont go away even though I adressed your concerns/replied in the commit underneath them

**Comment by Deantwo** [comment]
Yeah deleted them, will look at the correct place and comment there instead now.

**Comment by TheVekter** [comment]
Doris, I should be able to plop it down on the map once it's merged. Hit me up in IRC and I'll do it.

File: code/game/machinery/computer/communications.dm [hunk 1]
```diff
@@ -230,6 +230,19 @@ var/const/CALL_SHUTTLE_REASON_LENGTH = 12
 			src.emagged = 0
 			src.updateDialog()
 
+		if("nukerequest") //When there's no other way
+			if(src.authenticated==2)
+				if(CM.cooldownLeft())
+					usr << "Arrays recycling. Please stand by."
+					return
+				var/input = stripped_input(usr, "Please enter the reason for requesting the nuclear self-destruct codes. Misuse of the nuclear request system will not be tolerated under any circumstances.  Transmission does not guarantee a response.", "To abort, send an empty request.","")
```
Comment (Deantwo) [inline code/game/machinery/computer/communications.dm]: Please remove the ```"To abort, send an empty request."```, this is what the cancel button on the input window is for.
Replace with ```"Request Nuclear Authentication Codes"``` maybe as this is simply what is stated in the title of the input window.

**Comment by Jordie0608** [comment]
Changelog.

**Comment by Alek2ander** [comment]
Can this be a "2 IDs swiped at the same time" sort of thing? Self-destruction is a big deal yo.

**Comment by Alek2ander** [comment]
Why not both comms and authentification devices? Make the codes be requestable only in red alert.

**Comment by Dorsisdwarf** [comment]
Because I polled sybil, basil, #coderbus, and #tgtstation13, and 'captain ID at the comms console' was the clear majority, alek2ander.

**Comment by Dorsisdwarf** [comment]
Merge conflicts whyyyyyy

**Comment by Dorsisdwarf** [comment]
I'll just put up with manually adminning myself. it's not worth the hassle.

File: html/changelogs/Dorsisdwarf-PR-7476.yml.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# imageadd
+# imagedel
+# spellcheck (typo fixes)
+# experiment
+# tgs (TG-ported fixes?)
+#################################
+# Your name.
+author: Dorsidwarf
+# Optional: Remove this file after generating master changelog. Useful for PR changelogs that won't get used again.
+#delete-after: True
```
Comment (Deantwo) [inline html/changelogs/Dorsisdwarf-PR-7476.yml.txt]: Remove the ```#```.

File: html/changelogs/Dorsisdwarf-PR-7476.yml.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+# Any changes you've made. See valid prefix list above.
+# INDENT WITH TWO SPACES. NOT TABS. SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, this gets changed to [] after reading. Just remove the brackets when you add new shit.
+# Please surround your changes in single (') or double quotes ("). It works without them, but if you use certain characters it screws up compiling. The quotes will not show up in the changelog.
+changes:
+- rscadd: "Added the ability to call Central Command for the nuclear authentication codes. This requires a captain-level ID and admin approval, and warns the crew."
```
Comment (Deantwo) [inline html/changelogs/Dorsisdwarf-PR-7476.yml.txt]: Is this ```- ```?
Ot has to be ```  ```, aka two spaces.

File: code/game/gamemodes/nuclear/nuclearbomb.dm [hunk 4]
```diff
@@ -17,6 +17,15 @@ var/bomb_set
 	use_power = 0
 	var/previous_level = ""
 	var/lastentered = ""
+	var/immobile = 0 //Not all nukes should be moved
+
+/obj/machinery/nuclearbomb/selfdestruct
+	name = "station self-destruct"
```
Comment (Deantwo) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: At ``` console``` or ``` controller``` or something to the end of the name?
Just an idea.

Comment (Dorsisdwarf) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: How about "Interface"?

Comment (Deantwo) [inline code/game/gamemodes/nuclear/nuclearbomb.dm]: Yeah was thinking that too.
Or ``` terminal```.

File: html/changelogs/Dorsisdwarf-PR-7476.yml.txt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+author: Dorsidwarf
+# Optional: Remove this file after generating master changelog. Useful for PR changelogs that won't get used again.
+delete-after: True
+# Any changes you've made. See valid prefix list above.
+# INDENT WITH TWO SPACES. NOT TABS. SPACES.
+# SCREW THIS UP AND IT WON'T WORK.
+# Also, this gets changed to [] after reading. Just remove the brackets when you add new shit.
+# Please surround your changes in single (') or double quotes ("). It works without them, but if you use certain characters it screws up compiling. The quotes will not show up in the changelog.
+changes:
+-  rscadd: "Added the ability to call Central Command for the nuclear authentication codes. This requires a captain-level ID and admin approval, and warns the crew."
```
Comment (Deantwo) [inline html/changelogs/Dorsisdwarf-PR-7476.yml.txt]: Ok sorry To should have looked at the example, this is still wrong.
It has to be ```  - ```, aka: space, space, dash, space.
So weird.

**Comment by Dorsisdwarf** [comment]
Okay, that should be the changelog an name issues fixed.

**Comment by Dorsisdwarf** [comment]
Closed due to that 23 commits.

**Comment by Deantwo** [comment]
> Closed due to that 23 commits.

Why would the number of commits matter?

**Comment by RemieRichards** [comment]
@Deantwo
It clogs up the Commit log like mad.
Something like this can get away with being 1 or maybe 2 commits, not <i>23</i>

**Comment by Dorsisdwarf** [comment]
Okay, new PR is looking good.

**Pull request closed without merging**